internal/blobtesting/map.go

```diff
@@ -100,7 +115,13 @@ func (s *mapStorage) PutBlob(ctx context.Context, id blob.ID, data blob.Bytes, o
 
 	data.WriteTo(&b)
 
+	if s.limit >= 0 && s.totalBytes+int64(b.Len()) > s.limit {
+		return errors.Errorf("exceeded limit, unable to add %v bytes, currently using %v/%v", b.Len(), s.totalBytes, s.limit)
```

Comment:
We should not update the `s.keyTime` if the above check fails. This will mess up the key timestamps.

Reply:
done